File: srav_to_txt.py
import getopt
import sys

recLength = 170
debug = False


def usage():
    print("""
usage: SRAV_to_TXT [options] in.txt out.txt
                options:
                                -d            Turn on debug messages
---
"""
          )
    exit(2)


try:
    opts, args = getopt.getopt(sys.argv[1:], 'd')
    debug = dict(opts).setdefault('d', False)
except getopt.GetoptError as err:
    usage()

if len(args) != 2:
    usage()

try:
    IN = open(args[0], 'rb')
    OUT = open(args[1], 'w')

    while True:
        d = IN.read(recLength)
        if not d: break
        d = d.decode('cp037', 'replace')
        if d[0:2] == '47':
            OUT.write(d[19:140 + 19].rstrip() + '\n')

except FileNotFoundError as e:
    print("File {} does not exist".format(e.filename))

File: srav_to_txt.pl
#!/usr/bin/perl -w
use strict;
use POSIX;
use IO::File;
use Getopt::Std;
use Encode qw/encode decode/; 

my %opt;
getopts("d", \%opt);
my $recLen = 170+0;

usage(-1) unless @ARGV >= 2;

my $debug = defined($opt{'d'}) || 0;

open (IN, "<", $ARGV[0]) || die "Can't open input file $ARGV[0]\n";
binmode(IN);
open (OUT, ">", $ARGV[1]) || die "Can't open output file $ARGV[1]\n";
#binmode(OUT);

my $rec;

while(read(IN, $rec, $recLen ) > 0) {
                $rec = decode('cp37', $rec);
                if (substr($rec,0,2) eq '47') {
                                print OUT substr($rec,19,140), "\n";
                }

}
close(IN);
close(OUT);

sub usage {
                print << "---"
usage: SRAV_to_TXT [options] in.txt out.txt
                options:
                                -d            Turn on debug messages
---
;
                exit(-1);
}
